### PR TITLE
Update aot profile

### PR DESCRIPTION
**Pull request by PureWeen**
#### Before

```
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +659ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +687ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +659ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +659ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +658ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +649ms
```

#### After
```
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +607ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +600ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +617ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +624ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +633ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +631ms
ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +604ms
```

### CODE REVIEW

File: .nuspec/maui-blazor.aotprofile.txt [hunk 1]
```diff
@@ -6567,6 +6643,62 @@ Methods:
 	void System.Text.Json.WriteStack:EnsurePushCapacity ()
 	void System.Text.Json.WriteStack:Pop (bool)
 	void System.Text.Json.WriteStack:Push ()
+	void System.Text.RegularExpressions.Regex:.cctor ()
+	void System.Text.RegularExpressions.Regex:.ctor (string,System.Globalization.CultureInfo)
```
Comment (jonathanpeppers) [inline .nuspec/maui-blazor.aotprofile.txt:6647]: Do we know which commit started using Regex? this in itself might slow down startup some.

File: .nuspec/maui.aotprofile.txt [hunk 2]
```diff
@@ -1113,10 +1116,16 @@ Methods:
 	int System.Private.CoreLib.Interop/Sys:PRead (System.Runtime.InteropServices.SafeHandle,byte*,int,long)
 	int System.Random:Next (int,int)
 	int System.Random/XoshiroImpl:Next (int,int)
+	int System.Reflection.Emit.DynamicMethod:AddRef (object)
+	int System.Reflection.Emit.DynamicMethodTokenGenerator:GetToken (System.Reflection.MemberInfo,bool)
+	int System.Reflection.Emit.ILExceptionInfo:LastClauseType ()
+	int System.Reflection.Emit.ILGenerator:get_ILOffset ()
```
Comment (jonathanpeppers) [inline .nuspec/maui.aotprofile.txt:1122]: Hmm, we should see why these appeared. Nothing should use System.Reflection.Emit on startup now.

Do you see a log message like this one?

https://github.com/xamarin/xamarin-android/commit/f5ae4ca41caa7873443ebebe1d0a53595ff6495b

If you have `adb shell setprop debug.mono.log default,timing,assembly,mono_log_level=debug,mono_log_mask=aot` (assembly as one of the categories), it should log this message.

Comment (PureWeen) [reply .nuspec/maui.aotprofile.txt:1122]: `2022-04-15 12:50:41.557 30455-30455/? D/monodroid-assembly: Falling back to System.Reflection.Emit for delegate type '_JniMarshal_PPJ_Z': Boolean n_ContainsItem_J(IntPtr, IntPtr, Int64)`

File: src/Workload/Shared/FrameworkList.targets [hunk 3]
```diff
@@ -3,8 +3,6 @@
   <ItemGroup>
     <_FrameworkListFile Condition=" !$(MSBuildProjectName.Contains('.Runtime')) " Include="$(IntermediateOutputPath)FrameworkList.xml" />
     <_FrameworkListFile Condition=" !$(MSBuildProjectName.Contains('.Ref')) " Include="$(IntermediateOutputPath)RuntimeList.xml" />
-    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Packaging"  Version="6.0.0-beta.21169.1" PrivateAssets="all" />
```
Comment (eerhardt) [inline src/Workload/Shared/FrameworkList.targets:6]: This seems unrelated. Can it be a separate, independent PR?

File: .nuspec/maui.aotprofile.txt (hunk 2 above)
Comment (eerhardt) [reply .nuspec/maui.aotprofile.txt:1122]: Sounds like that should be fixed in xamarin/android. Just like https://github.com/xamarin/xamarin-android/pull/6855

File: src/Workload/Shared/FrameworkList.targets (hunk 3 above)
Comment (PureWeen) [reply src/Workload/Shared/FrameworkList.targets]: Yea this was needed so I could build the workload
I rebased now that https://github.com/dotnet/maui/pull/6130 merged

File: .nuspec/maui-blazor.aotprofile.txt (hunk 1 above)
Comment (jonathanpeppers) [reply .nuspec/maui-blazor.aotprofile.txt:6647]: Ok, yeah if it is only in the `maui-blazor` template, it might be something in ASP.NET.

Maybe we don't worry about this one for now.

Comment (PureWeen) [reply .nuspec/maui-blazor.aotprofile.txt:6647]: I'm not really sure on this one.

Maybe this PR created a path that now executes the RegEx here?
https://github.com/dotnet/maui/commit/c14e363fa8bc90ef82532ea795f523156a5d0205

We don't really make use of RegEx much on the Android side of things. 
https://github.com/dotnet/maui/search?q=RegEx

Comment (jonathanpeppers) [reply .nuspec/maui-blazor.aotprofile.txt:6647]: It could also be: https://github.com/dotnet/maui/commit/87af742bb9a6ed17cede4e4806348e51c5693cc1

I'll get a speedscope of latest Maui and look for the stack trace.

Comment (jonathanpeppers) [reply .nuspec/maui-blazor.aotprofile.txt:6647]: There does seem to be some cost in loading Regex the first time, at:

https://github.com/dotnet/maui/blob/e9852b3cdbe8445737fbcb3124385652c18e6b5f/src/BlazorWebView/src/SharedSource/StaticContentHotReloadManager.cs#L20

image

[maui-blazor.zip](https://github.com/dotnet/maui/files/8498033/maui-blazor.zip)

This was kind of a franken build with https://github.com/xamarin/xamarin-android/pull/6931 and dotnet/maui 6.0.300-rc.2.5465. I had to turn off AOT, because I was getting a crash.

**Comment by jonathanpeppers** [comment]
I updated these here: https://github.com/dotnet/maui/pull/6238

System.Reflection.Emit is gone now on that one. 👍
